### PR TITLE
Consents Management redux integration and UI enhancements

### PR DESCRIPTION
**Pull request by brionmario**
Screen Shot 2019-08-19 at 5 46 47 PM
Screen Shot 2019-08-19 at 5 49 33 PM
Screen Shot 2019-08-26 at 3 16 32 PM

### CODE REVIEW

File: apps/user-portal/src/actions/consent-management.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@
  * under the License.
  */
 
-import { ConsentInterface, ConsentReceiptInterface, ConsentState } from "../models/consents";
```
Comment (jeradrutnam) [inline apps/user-portal/src/actions/consent-management.ts]: Keep it as `consents`

File: apps/user-portal/src/actions/consent-management.ts [hunk 2]
```diff
@@ -311,132 +311,132 @@ export const updateRevokedClaimIds = (claimIds: number[]): UpdateRevokedClaimIds
  *
  * @type {string}
  */
-export const SHOW_CONSENTS_EDIT_VIEW = "SHOW_CONSENTS_EDIT_VIEW";
+export const SHOW_CONSENT_EDIT_VIEW = "SHOW_CONSENT_EDIT_VIEW";
```
Comment (jeradrutnam) [inline apps/user-portal/src/actions/consent-management.ts]: Move actionTypes to a separate file. May be we can create a `types` folder inside `actions` folder.

File: apps/user-portal/src/components/icon.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import cx from "classnames";
```
Comment (jeradrutnam) [inline apps/user-portal/src/components/icon.tsx]: use as `classNames` instead to keep the consistent

File: apps/user-portal/src/components/ui.tsx [hunk 4]
```diff
@@ -52,7 +60,8 @@ export const Title = (props: TitleProps) => {
 
     return (
         <div className={classNames(classes, "product-title")} style={style}>
-            <Logo /> <h1 className={classNames(classes, "product-title-text")} style={style}>{TitleText}</h1>
+            <Logo size="mini"/>
```
Comment (jeradrutnam) [inline apps/user-portal/src/components/ui.tsx]: Remove the hard-coded `size` value

File: apps/user-portal/src/helpers/ui.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+/**
```
Comment (jeradrutnam) [inline apps/user-portal/src/helpers/ui.ts]: Lets rename the file as `image.ts`

File: apps/user-portal/src/middlewares/consent-management.ts [hunk 6]
```diff
@@ -56,27 +56,33 @@ const handleFetchConsentedApps = ({ dispatch, getState }) => (next) => (action)
...
+    AuthenticateSessionUtil.getAccessToken()
+        .then((token) => {
+            const state: ConsentState = getState().consentManagement.consentState;
+            const requestConfig: HttpRequestConfig = {
+                data: {
+                    piiPrincipalId: AuthenticateSessionUtil.getSessionParameter(AuthenticateUserKeys.USERNAME),
+                    state
+                },
+                dispatcher: FETCH_CONSENTED_APPS,
+                headers: {
```
Comment (jeradrutnam) [inline apps/user-portal/src/middlewares/consent-management.ts]: Try use the headers from the common util where it's possible

File: apps/user-portal/src/middlewares/consent-management.ts [hunk 7]
```diff
@@ -159,23 +165,29 @@ const handleFetchConsentReceipt = ({ dispatch }) => (next) => (action) => {
...
+            const id: string = action.payload;
+            const requestConfig: HttpRequestConfig = {
+                dispatcher: FETCH_CONSENT_RECEIPT,
+                headers: {
+                    "Accept": "application/json",
+                    "Access-Control-Allow-Origin": CLIENT_HOST,
+                    "Authorization": `Bearer ${ token }`,
+                    "Content-Type": "application/json"
+                },
+                method: "get",
```
Comment (jeradrutnam) [inline apps/user-portal/src/middlewares/consent-management.ts]: Keep the method on consistent case. Lowercase or Uppercase everywhere

File: modules/theme/src/definitions/globals/product.less [hunk 8]
```diff
@@ -73,6 +75,140 @@ body {
...
     }
+
+    .icon-wrapper {
+        background-color: @iconWrapperBgColor;
+        overflow: hidden;
+        display: flex;
+        text-align: center;
+        align-items: center;
+        justify-content: center;
+        &.inline {
```
Comment (jeradrutnam) [inline modules/theme/src/definitions/globals/product.less]: Keep a new-line after attributes finishes and the start of class continuation

File: modules/theme/src/themes/default/assets/images/default-app-icon.svg [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+ * under the License.
+-->
+
+<svg class="icon" version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
+	 width="14px" height="14px" viewBox="0 0 374 374" enable-background="new 0 0 374 374" xml:space="preserve">
+<g>
+	<path class="primary" d="M184.832,121.211c-17.401,0-31.56,14.157-31.56,31.561c0,5.243,1.285,10.191,3.557,14.546
+		c5.269,10.103,15.844,17.016,28.003,17.016c17.403,0,31.562-14.159,31.562-31.562C216.395,135.369,202.235,121.211,184.832,121.211
+		z"/>
+	<path class="primary" d="M326.207,77.847L188.32,4.374c-2.046-1.092-4.498-1.077-6.528,0.038L47.72,77.885
```
Comment (jeradrutnam) [inline modules/theme/src/themes/default/assets/images/default-app-icon.svg]: Change to `class="secondary"`

File: modules/theme/src/definitions/globals/product.less [hunk 10]
```diff
@@ -73,6 +75,140 @@ body {
...
+            &.square {
+                svg.icon {
+                    height: 960px;
+                }
+            }
+        }
+        &.two-tone {
+            svg.icon {
+                path.primary {
+                    fill: @primaryColor;
```
Comment (jeradrutnam) [inline modules/theme/src/definitions/globals/product.less]: Move it to a new variable in `product.less` and use here and below. ex:- `@iconWrapperTwoTonePrimaryColor`

File: modules/theme/src/definitions/globals/product.less [hunk 11]
```diff
@@ -73,6 +75,140 @@ body {
...
+        &.default {
+            svg.icon {
+                path {
+                    fill: @iconDefaultBgColor;
+                }
+            }
+        }
+        &.mini {
+            svg.icon {
+                width: 35px;
```
Comment (jeradrutnam) [inline modules/theme/src/definitions/globals/product.less]: Move sizes to a variable in `global.variables` file and use

File: apps/user-portal/src/components/ui.tsx [hunk 12]
```diff
@@ -102,3 +102,29 @@ export const HomeTileIcon = (props: HomeTileIconImageProps) => {
...
+    let marginLeft = 0;
+    if (spaced && spaced === "right") {
+        marginRight = 25;
+    } else if (spaced && spaced === "left") {
+        marginLeft = 25;
+    }
+
+    return (
+        <div
+            className="generic-app-icon"
```
[the remark this answers is not in the document]
Comment (brionmario) [reply apps/user-portal/src/components/ui.tsx]: A reusable icon component was created as suggested to address this issue.

File: modules/theme/src/definitions/views/user-portal.less [hunk 13]
```diff
@@ -22,4 +22,19 @@
 
 @import (multiple) '../../wso2.config';
 
+.generic-app-icon {
+  background-color: @lightGray;
+  padding: 5px;
```
[the remark this answers is not in the document]
Comment (brionmario) [reply modules/theme/src/definitions/views/user-portal.less]: It was set to 2 by mistake. I've made it 4.

File: modules/theme/src/definitions/views/user-portal.less [hunk 14]
```diff
@@ -22,4 +22,19 @@
 
 @import (multiple) '../../wso2.config';
 
+.generic-app-icon {
+  background-color: @lightGray;
```
[the remark this answers is not in the document]
Comment (brionmario) [reply modules/theme/src/definitions/views/user-portal.less]: A variable called ```@iconWrapperBgColor``` was created in ```product.variables``` file in ```@wso2is/theme``` as suggested.

File: modules/theme/src/definitions/views/user-portal.less [hunk 15]
```diff
@@ -22,4 +22,19 @@
...
+  padding: 5px;
+  text-align: center;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border-width: 1px;
+  border-style: solid;
+  border-color: @darkGray;
+  border-image: initial;
+  border-radius: 3px;
```
[the remark this answers is not in the document]
Comment (brionmario) [reply modules/theme/src/definitions/views/user-portal.less]: Hard coded value was replaced with ```@defaultBorderRadius``` variable.

File: apps/user-portal/src/actions/consent-management.ts (hunk 2 above)
Comment (brionmario) [reply apps/user-portal/src/actions/consent-management.ts]: A separate folder was created inside actions directory and corresponding ```.ts``` files were created and the action types were moved in to those files.

File: apps/user-portal/src/components/ui.tsx (hunk 4 above)
Comment (brionmario) [reply apps/user-portal/src/components/ui.tsx]: size attribute was removed.

File: apps/user-portal/src/middlewares/consent-management.ts (hunk 7 above)
Comment (brionmario) [reply apps/user-portal/src/middlewares/consent-management.ts]: Method types were made uppercase.

File: modules/theme/src/definitions/globals/product.less (hunk 11 above)
Comment (brionmario) [reply modules/theme/src/definitions/globals/product.less]: Sizes have been moved to ```product.variables``` file.

File: apps/user-portal/src/middlewares/consent-management.ts (hunk 6 above)
Comment (jeradrutnam) [reply apps/user-portal/src/middlewares/consent-management.ts]: We'll do this later

File: modules/theme/src/definitions/globals/product.less [hunk 16]
```diff
@@ -73,6 +75,142 @@ body {
...
+                height: auto;
+            }
+            &.square {
+                svg.icon {
+                    height: @iconSmall;
+                }
+            }
+        }
+        &.medium {
+             svg.icon {
```
Comment (jeradrutnam) [inline modules/theme/src/definitions/globals/product.less]: There is an additional indentation

**Pull request merged**